pulsar-broker/src/main/java/org/apache/pulsar/PulsarClusterMetadataSetup.java

```diff
@@ -244,7 +244,8 @@ public static void main(String[] args) throws Exception {
         // Format BookKeeper ledger storage metadata
         if (arguments.existingBkMetadataServiceUri == null && arguments.bookieMetadataServiceUri == null) {
             ServerConfiguration bkConf = new ServerConfiguration();
-            bkConf.setMetadataServiceUri("metadata-store:" + metadataStoreUrlNoIdentifer);
+            bkConf.setDelimiterParsingDisabled(true);
```

Comment:
why setDelimiterParsingDisabled ?
are we setting it anywhere else  ?

Reply:
BK setMetadataServiceUri will set a ArrayList, If a string with comma delimiter. eg zk1:2181,zk2:2181/test will get result ["zk1:2181", "zk2:2181/test"]. getMetadataServiceUri only return the first part in list zk1:2181.   @eolivelli 
Bookeeper also has this problem, see https://github.com/apache/bookkeeper/issues/2900.
Without   setDelimiterParsingDisabled,  zk connection string with comma is not supported.

Reply:
I think apache/bookkeeper#2900 is not the same as the problem here. It's because the 
`ServerConfiguration`(extend `CompositeConfiguration`) can't process a comma-separated parameter read from `bk_server.conf`(if you set `zk1:2181,zk2:2181/test` in bk_server.conf, you only get the `zk1:2181` in `ServerConfiguration`).

But here just `ServerConfiguration.setMetadataServiceUri()`, why  we will get `["zk1:2181", "zk2:2181/test"].` ?

Reply:
if we add this option here (bkConf.setDelimiterParsingDisabled(true)), we should set it everywhere, anytime we create a ServerConfiguration object, and this is very hard to maintain.

Are you sure that you are able to run a full Pulsar cluster=  if we are not setting the same option in the Bookie it won't start, the same for the Auto Recovery process probably

Reply:
if we are not setting the same option in the Bookie, `metadataServiceUri` in bookkeeper.conf  don't support ZK connection string with comma. Only support string with  semicolon @eolivelli

Reply:
> I think [apache/bookkeeper#2900](https://github.com/apache/bookkeeper/issues/2900) is not the same as the problem here. [apache/bookkeeper#2900](https://github.com/apache/bookkeeper/issues/2900) caused by the `ServerConfiguration`(extend `CompositeConfiguration`) can't process a comma-separated parameter read from `bk_server.conf`(if you set `zk1:2181,zk2:2181/test` in bk_server.conf, you only get the `zk1:2181` in `ServerConfiguration`).
> 
> But here just `ServerConfiguration.setMetadataServiceUri()`, why we will get `["zk1:2181", "zk2:2181/test"].` ?

@aloyszhang   You can try `bin/pulsar initialize-cluster-metadata --cluster test --metadata-store  zk1:2181,zk2:2181/test --configuration-metadata-store zk1:2181,zk2:2181/test   --web-service-url http://localhost:8080/  --broker-service-url pulsar://localhost:6650`

Reply:
@gaozhangmin Have checked this, either `setMetadataServiceUri()` or read from a file both lose the content after the first comma.